### PR TITLE
Use `.effekt.md` for literate Effekt code

### PR DESCRIPTION
**Pull request by marzipankaiser**
This mirrors https://github.com/effekt-lang/effekt-vscode/pull/14, i.e.:
- Use `.effekt.md` instead of plain `.md` for literate Effekt code.
- Also changes the treatment of markdown files such that code fences that are explicitly labelled with something other than   
  `effekt` (plus optional spaces) are ignored.
  I.e. valid (opening and closing) fence delimiters:
  ~~~
  ```
  ```effekt
  ``` effekt
  ```effekt :mymodifier
  ```effekt:mymodifier
  ```effekt timeout=12
  ~~~
  Ignored delimiters:
  ~~~
  ```tex
  ```myfancyotherlanguage
  ```effektcode
  ```eff
  ~~~

### CODE REVIEW

**Review by jiribenes — APPROVED** [review]
LGTM :shipit:

**Comment by marzipankaiser** [comment]
Note: In some of the case studies, we use something like
~~~
```
// output
```
~~~

To give the output of a function etc.
Using this PR, we could instead do:
~~~
```text
output
```
~~~
directly.

**Review by b-studios — APPROVED** [review]

File: effekt/shared/src/main/scala/effekt/context/ModuleDB.scala [hunk 1]
```diff
@@ -43,7 +43,7 @@ trait ModuleDB { self: Context =>
     }
 
   private def stripSuffix(path: String): String =
-    path.stripSuffix(".effekt").stripSuffix(".md")
+    path.stripSuffix(".md").stripSuffix(".effekt")
```
[the remark this answers is not in the document]
Comment (marzipankaiser) [reply effekt/shared/src/main/scala/effekt/context/ModuleDB.scala:46]: Yes :)

File: effekt/shared/src/main/scala/effekt/util/Source.scala [hunk 2]
```diff
@@ -10,17 +10,22 @@ import kiama.util.Source
  */
 case class MarkdownSource(source: Source) extends Source {
 
-  val fenceLine = """^```[^`\n]*$""".r
+  val fenceLine = """^```\s*(effekt)?\s*((\s|:)[^`]*)?$""".r
```
[the remark this answers is not in the document]
Comment (marzipankaiser) [reply effekt/shared/src/main/scala/effekt/util/Source.scala:13]: It should. It does support the ones here:
https://github.com/effekt-lang/effekt/blob/c4febaa216a1054cb39cfcce886203f7a65794b5/examples/casestudies/ad.md?plain=1#L16-L23
(which were the motivation behind allowing ```(\s|:)[^`]*)?``` there)


So I would be inclined to say yes, though I did not specifically test for them.

**Pull request merged**